Invert CgroupDevicesEnabled condition

vdemeester opened the pull request:
As a follow-up to #22091 and to follow @justincormack comment https://github.com/docker/docker/pull/22091#discussion-diff-64142769 🐼 :

> It seems better to test where it's required than listing everywhere it is not required.

/cc @thaJeztah @runcom @LK4D4 @amitkris

🐸

Signed-off-by: Vincent Demeester vincent@sbr.pm

thaJeztah commented:
LGTM, thanks!

LK4D4 commented on daemon/daemon.go:
```diff
@@ -795,7 +795,7 @@ func NewDaemon(config *Config, registryService *registry.Service, containerdRemo
 	// Check if Devices cgroup is mounted, it is hard requirement for container security,
 	// on Linux/FreeBSD.
-	if runtime.GOOS != "windows" && runtime.GOOS != "solaris" && !sysInfo.CgroupDevicesEnabled {
+	if (runtime.GOOS == "linux" || runtime.GOOS == "freebsd") && !sysInfo.CgroupDevicesEnabled {
```
This will be always true on freebsd. There is no cgroups there.

justincormack replied:
Yes it is odd this was here, but it must be incorrect.

vdemeester replied:
ah, then `runtime.GOOS == "freebsd"` wouldn't even be needed :stuck_out_tongue:

vdemeester replied:
ping @kvasdopil just to be sure :sweat_smile:

vdemeester committed:
Invert CgroupDevicesEnabled condition

It feels better to test where it's required than listing everywhere it
is not required.

Signed-off-by: Vincent Demeester <vincent@sbr.pm>

vdemeester commented:
@LK4D4 @justincormack update by removing the Freebsd part of the condition

runcom commented:
LGTM

thaJeztah commented:
LGTM
